libraries/stdlib/src/kotlin/text/Char.kt

```diff
@@ -8,6 +8,72 @@
+        throw IllegalArgumentException("Invalid Char code: $code")
+    }
+    return code.toChar()
+}
+
+/**
+ * Returns the code of this Char.
+ *
+ * Code of a Char is the value it was constructed with, and the UTF-16 code unit corresponding to this Char.
+ */
+public inline val Char.code: Int get() = this.toInt()
+
+/**
+ * Returns the numeric value of the digit that this Char represents in the specified [radix].
+ * Throws an exception if the [radix] is not in the range `2..36` or if this Char is not a valid digit in the specified [radix].
+ *
+ * A Char is considered to represent a digit in the specified [radix] if at least one of the following is true:
+ *  - [isDigit] is `true` for the Char and the Unicode decimal digit value of the character is less than the specified [radix]. In this case the decimal digit value is returned.
+ *  - The Char is one of the uppercase Latin letters 'A' through 'Z' and its [code] is less than `radix + 'A'.code - 10`. In this case, `this.code - 'A'.code + 10` is returned.
+ *  - The Char is one of the lowercase Latin letters 'a' through 'z' and its [code] is less than `radix + 'a'.code - 10`. In this case, `this.code - 'a'.code + 10` is returned.
+ */
+public fun Char.digitToInt(radix: Int = 10): Int {
+    TODO("Implement using digits table from CharCategory PR")
+}
+
+/**
+ * Returns the numeric value of the digit that this Char represents in the specified [radix], or `null` if this Char is not a valid digit in the specified [radix].
+ * Throws an exception if the [radix] is not in the range `2..36`.
+ *
+ * A Char is considered to represent a digit in the specified [radix] if at least one of the following is true:
+ *  - [isDigit] is `true` for the Char and the Unicode decimal digit value of the character is less than the specified [radix]. In this case the decimal digit value is returned.
+ *  - The Char is one of the uppercase Latin letters 'A' through 'Z' and its [code] is less than `radix + 'A'.code - 10`. In this case, `this.code - 'A'.code + 10` is returned.
+ *  - The Char is one of the lowercase Latin letters 'a' through 'z' and its [code] is less than `radix + 'a'.code - 10`. In this case, `this.code - 'a'.code + 10` is returned.
+ */
+public fun Char.digitToIntOrNull(radix: Int = 10): Int? {
+    TODO("Implement using digits table from CharCategory PR")
+}
+
+/**
+ * Returns the Char that represents this numeric digit value in the specified [radix].
+ * Throws an exception if the [radix] is not in the range `2..36` or if this value is not less than the specified [radix].
+ *
+ * If this value is less than `10`, the decimal digit Char with code `'0'.code + this` is returned.
+ * Otherwise, the uppercase Latin letter with code `'A'.code + this - 10` is returned.
+ */
+public fun Int.digitToChar(radix: Int = 10): Char {
+    if (radix !in 2..36) {
+        throw IllegalArgumentException("Invalid radix: $radix. Valid radix values are in range 2..36")
+    }
+    if (this >= radix) {
```

Comment:
What about negatives?

Reply:
The check above (`radix !in 2..36`) excludes negative radix

Reply:
I mean negative digit

Reply:
done